lib/pos_mempool.go

```diff
@@ -0,0 +1,490 @@
+package lib
+
+import (
+	"github.com/dgraph-io/badger/v3"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"path/filepath"
+	"strings"
+	"sync"
+)
+
+type PosMempoolStatus int
+
+const (
+	PosMempoolStatusRunning PosMempoolStatus = iota
+	PosMempoolStatusNotRunning
+)
+
+type Mempool interface {
+	Start() error
+	Stop()
+	IsRunning() bool
+	AddTransaction(txn *MsgDeSoTxn) error
+	RemoveTransaction(txnHash *BlockHash) error
+	GetTransaction(txnHash *BlockHash) *MsgDeSoTxn
+	GetTransactions() []*MsgDeSoTxn
+	GetIterator() MempoolIterator
+	UpdateLatestBlock(blockView *UtxoView, blockHeight uint64)
+	UpdateGlobalParams(globalParams *GlobalParamsEntry)
+}
+
+type MempoolIterator interface {
+	Next() bool
+	Value() (*MsgDeSoTxn, bool)
+	Initialized() bool
+}
+
+// PosMempool is used by the node to keep track of uncommitted transactions. The main responsibilities of the PosMempool
+// include addition/removal of transactions, back up of transaction to database, and retrieval of transactions ordered
+// by Fee-Time algorithm. More on the Fee-Time algorithm can be found in the documentation of TransactionRegister.
+type PosMempool struct {
+	sync.RWMutex
```

Comment:
Avoid the use of embedding in Go. Better to call this something like `mtx sync.RWMutex`

Reasons to avoid embedding:
* A lot of devs aren't used to it. Even I actually had to do a double-take when I saw it thinking "is this a new feature?"
* When you see PosMempool.something in various places, it makes it more annoying to figure out where it came from because you can't just look at the struct and see it.
* If the definition of the thing you're embedding changes, it can have unexpected behaviors, adding and removing fields from your struct randomly.

Reply:
Hm, I disagree with this one. I think embeddings make sense when they simplify the syntax and meaningfully augment the functionality to the type. In the case of mutex, imo embedding is the preferred path, since Lock/Unlock functionality is common enough to meaningfully augment PosMempool type.

In terms of the prevelance of the embedding pattern, I actually started using the mutex embedding after seeing it in a bunch of blockchain repos. So I think the readers of our code will be comfortable with this embedding.